src/legacy/core_plugins/metrics/public/components/panel_config/timeseries.js

```diff
@@ -158,16 +200,25 @@ class TimeseriesPanelConfigUi extends Component {
               <EuiFlexItem>
                 <EuiFormRow
                   id={htmlId('panelFilter')}
-                  label={(<FormattedMessage
-                    id="tsvb.timeseries.optionsTab.panelFilterLabel"
-                    defaultMessage="Panel filter"
-                  />)}
+                  label={
+                    <FormattedMessage
+                      id="tsvb.timeseries.optionsTab.panelFilterLabel"
+                      defaultMessage="Panel filter"
+                    />
+                  }
                   fullWidth
                 >
-                  <EuiFieldText
-                    onChange={handleTextChange('filter')}
-                    value={model.filter}
-                    fullWidth
+                  <QueryBar
+                    query={{
+                      language: model.filter.language ? model.filter.language : 'kuery',
```

Comment:
I see that you use 'kuery' only here. In other places you use uiSettingsQueryLanguage. Is it correct?

Reply:
I will change it.